core/domain/opportunity_services.py

```diff
@@ -407,6 +407,147 @@ def update_opportunities_with_new_topic_name(topic_id, topic_name):
+            cursor: str or None. A query cursor pointing to the next
+                batch of results. If there are no more results, this might
+                be None.
+            more: bool. If True, there are (probably) more results after
+                this batch. If False, there are no further results after
+                this batch.
+    """
+    page_size = feconf.OPPORTUNITIES_PAGE_SIZE
+    skill_opportunity_models, cursor, more = (
+        opportunity_models.SkillOpportunityModel
+        .get_skill_opportunities(page_size, cursor))
+    opportunities = []
+    for skill_opportunity_model in skill_opportunity_models:
+        skill_opportunity = (
+            get_skill_opportunity_from_model(
+                skill_opportunity_model))
+        opportunities.append(skill_opportunity.to_dict())
+    return opportunities, cursor, more
+
+
+def create_skill_opportunity(skill_opportunity):
+    """Creates a skill opportunity domain object as a SkillOpportunityModel
+    entity in the datastore.
+
+    Args:
+        skill_opportunity: The skill opportunity domain object to be saved in
+            the datastore.
+    """
+    skill_opportunity_model = opportunity_models.SkillOpportunityModel(
+        id=skill_opportunity.id,
+        skill_description=skill_opportunity.skill_description,
+        question_count=skill_opportunity.question_count,
+    )
+    skill_opportunity_model.put()
+
+
+def update_skill_opportunity_skill_description(skill_id, new_description):
+    """Updates the skill_description of the SkillOpportunityModel with
+    new_description.
+
+    Args:
+        skill_id: str. The corresponding skill_id of the opportunity.
+        new_description: str. The new skill_description.
+    """
+    skill_opportunity_model = (
+        opportunity_models.SkillOpportunityModel.get(skill_id))
+    skill_opportunity_model.skill_description = new_description
+    skill_opportunity_model.put()
+
+
+def delete_skill_opportunities(skill_id):
+    """Deletes the SkillOpportunityModel corresponding to the supplied skill_id.
+
+    Args:
+        skill_id: str. The skill_id corresponding to the to-be-deleted
+            SkillOpportunityModel.
+    """
+    skill_opportunity_model = (
+        opportunity_models.SkillOpportunityModel.get(skill_id))
+    opportunity_models.SkillOpportunityModel.delete(skill_opportunity_model)
+
+
+def update_skill_opportunity_question_counts(skill_ids, delta):
+    """Increments question_count(s) of SkillOpportunityModel(s) with
+    corresponding skill_ids.
+
+    Args:
+        skill_ids: list(str). A list of skill_ids corresponding to
+            SkillOpportunityModel(s).
+        delta: int. The delta for which to update each question_count.
+    """
+    updated_models = _update_skill_opportunity_question_count(skill_ids, delta)
+    opportunity_models.SkillOpportunityModel.put_multi(updated_models)
+
+
+def compute_question_linked_skill_ids_update(old_skill_ids, new_skill_ids):
```

Comment:
Perhaps this should start with update_, otherwise it's not clear that it's actually modifying the datastore (as opposed to computing something transient).

Reply:
Done.